src/test/regress/expected/aggregate_support.out

```diff
@@ -85,46 +142,164 @@ select key, sum2(val) filter (where valf < 5), sum2_strict(val) filter (where va
-  1 |    4 |           4
-  2 |      |
-  3 |    4 |           4
-  4 |    6 |           6
-  5 |   10 |          10
-  6 |    8 |           8
-  7 |      |
-  8 |      |
-  9 |      |
- 10 |   16 |          16
- 11 |    0 |           0
+select id, sum2(distinct val), sum2_strict(distinct val), psum(distinct val, valf::int), psum_strict(distinct val, valf::int) from aggdata group by id order by id;
+ id | sum2 | sum2_strict | psum  | psum_strict
+---------------------------------------------------------------------
+  1 |    4 |           4 |    50 |          50
+  2 |      |             |     2 |           0
+  3 |    4 |           4 |    18 |          18
+  4 |    6 |           6 |    30 |          30
+  5 |   10 |          10 |    56 |          56
+  6 |    8 |           8 |   510 |         510
+  7 |      |             |     2 |           0
+  8 |      |             |     2 |           0
+  9 |      |             |     2 |           0
+ 10 |   16 |          16 | 17254 |       17254
+ 11 |    0 |           0 |     6 |           6
 (11 rows)
 
 -- ORDER BY unsupported
-select key, sum2(val order by valf), sum2_strict(val order by valf) from aggdata group by key order by key;
+select key, sum2(val order by valf), sum2_strict(val order by valf), psum(val, valf::int order by valf), psum_strict(val, valf::int order by valf) from aggdata group by key order by key;
 ERROR:  unsupported aggregate function sum2
 -- Test handling a lack of intermediate results
-select sum2(val), sum2_strict(val) from aggdata where valf = 0;
- sum2 | sum2_strict
+select sum2(val), sum2_strict(val), psum(val, valf::int), psum_strict(val, valf::int) from aggdata where valf = 0;
+ sum2 | sum2_strict | psum | psum_strict
 ---------------------------------------------------------------------
-    0 |
+    0 |             |    0 |           0
 (1 row)
 
 -- Test HAVING
-select key, stddev(valf) from aggdata group by key having stddev(valf) > 2 order by key;
- key |      stddev
+select key, stddev(valf)::numeric(10,5) from aggdata group by key having stddev(valf) > 2 order by key;
+ key | stddev
 ---------------------------------------------------------------------
-   1 | 6.43467170879758
+   1 | 6.43467
 (1 row)
 
-select key, stddev(valf) from aggdata group by key having stddev(val::float8) > 1 order by key;
- key |      stddev
+select key, stddev(valf)::numeric(10,5) from aggdata group by key having stddev(val::float8) > 1 order by key;
+ key | stddev
 ---------------------------------------------------------------------
-   2 | 1.01500410508201
+   2 | 1.01500
+(1 row)
+
+select key, corr(valf,valf+val)::numeric(10,5) from aggdata group by key having corr(valf,valf+val) < 1 order by key;
+ key |  corr
+---------------------------------------------------------------------
+   2 | 0.99367
+(1 row)
+
+select key, corr(valf,valf+val)::numeric(10,5) from aggdata group by key having corr(valf::float8,valf+val) < 1 order by key;
```

Comment:
I think, all these tests make a lot of sense, and they actually test what  we've implement in this PR. However, the majority of our bugs stem from using features in combinations. With that  regard, I manually tested the new feature with combination with other features.

I've used the  table definitions [here](https://gist.github.com/onderkalaci/97bd2ddfecef49751b4ccb93f498c5c1#file-main-sql-L2), and played with queries such as  the below. Feel free to pick  & adjust some of them to these tests:

```SQL
 
-- query with window functions, the agg. inside the window functions
 SELECT
    value_3, date_trunc('day', time) as time, rank() OVER my_win as my_rank
  FROM
    events_table
  GROUP BY
    value_3, date_trunc('day', time)
    WINDOW my_win AS (PARTITION BY regr_syy(event_type%10, value_2)::int ORDER BY count(*) DESC) 
 ORDER BY 1,2,3;

-- query with window functions, the agg. outside the window functions
 SELECT
    regr_syy(event_type%10, value_2)::int, value_3, date_trunc('day', time) as time, rank() OVER my_win as my_rank
  FROM
    events_table
  GROUP BY
    value_3, date_trunc('day', time)
    WINDOW my_win AS (PARTITION BY value_3 ORDER BY count(*) DESC) 
 ORDER BY 1,2,3;


-- query with only  ORDER BY
 SELECT
 	regr_syy(event_type%10, value_2)::int
 FROM
 	events_table
ORDER BY 1 DESC;

-- query with group by + target list + order by
 SELECT
 	count(*), regr_syy(event_type%10, value_2)::int
 FROM
 	events_table
 GROUP BY
 	value_3
ORDER BY 2 DESC;



-- query with group by + order by
 SELECT
 	count(*) 
 FROM
 	events_table
 GROUP BY
 	value_3
ORDER BY regr_syy(event_type%10, value_2)::int DESC;


-- query with basic join
SELECT
	regr_syy(u1.user_id, u2.user_id)
FROM 
	users_Table  u1, events_table u2
WHERE 
	u1.user_id = u2.user_id;


-- agg. with FILTER with columns
SELECT
regr_syy(u1.user_id, u2.user_id) FILTER (WHERE u1.value_1 < 5)
FROM users_Table  u1, events_table u2
WHERE u1.user_id = u2.user_id;

-- agg with filter and GROUP BY
SELECT
regr_syy(u1.user_id, u2.user_id) FILTER (WHERE (u1.value_1) < 5)
FROM users_Table  u1, events_table u2
WHERE u1.user_id = u2.user_id
GROUP BY u1.user_id;

-- agg. with FILTER with consts
SELECT
regr_syy(u1.user_id, u2.user_id) FILTER (WHERE '0300030' LIKE '%3%')
FROM users_Table  u1, events_table u2
WHERE u1.user_id = u2.user_id;

-- multiple aggs with filters
SELECT
regr_syy(u1.user_id, u2.user_id) FILTER (WHERE u1.value_1 < 5), regr_syy(u1.value_1, u2.value_2) FILTER (WHERE u1.user_id < 5)
FROM users_Table  u1, events_table u2
WHERE u1.user_id = u2.user_id;

-- query with WHERE false
SELECT
regr_syy(u1.user_id, u2.user_id) FILTER (WHERE u1.value_1 < 5), regr_syy(u1.value_1, u2.value_2) FILTER (WHERE u1.user_id < 5)
FROM users_Table  u1, events_table u2
WHERE 1=0;

-- a CTE forced to be planned recursively (via OFFSET 0)
WITH cte_1 AS 
(
	SELECT
	regr_syy(u1.user_id, u2.user_id) FILTER (WHERE u1.value_1 < 5), regr_syy(u1.value_1, u2.value_2) FILTER (WHERE u1.user_id < 5)
	FROM users_Table  u1, events_table u2
	WHERE u1.user_id = u2.user_id
	OFFSET 0
)
SELECT 
	* 
FROM 
	cte_1 
ORDER BY 
	1,2 LIMIT 5;

```